[JIT] SubgraphUtils: add a function for generating a string name for a given graph.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47255 [TensorExpr] Pick meaningful names for functions in TE codegen.
* #47254 [TensorExpr] CudaCodegen: restart counter for function names unique ID inside each codegen instantiation.
* **#47253 [JIT] SubgraphUtils: add a function for generating a string name for a given graph.**

The function simply goes over all aten nodes in the graph and
concatenates their names, truncating the final name to a given length.

Differential Revision: [D24698272](https://our.internmc.facebook.com/intern/diff/D24698272)

Commit:
[JIT] SubgraphUtils: add a function for generating a string name for a given graph.

The function simply goes over all aten nodes in the graph and
concatenates their names, truncating the final name to a given length.

[ghstack-poisoned]

bertmaher commented on torch/csrc/jit/passes/utils/subgraph_utils.cpp:
```diff
@@ -394,6 +394,31 @@ Node* createSingletonSubgraphAndUpdateAliasing(
+  }
+  std::stringstream truncated;
+  truncated << s.substr(0, maxlen);
+  truncated << "_" << std::hash<std::string>{}(s) << "_";
```
Any reason for the trailing `_`?

bertmaher commented on torch/csrc/jit/passes/utils/subgraph_utils.cpp:
```diff
@@ -394,6 +394,31 @@ Node* createSingletonSubgraphAndUpdateAliasing(
+    return s;
+  }
+  std::stringstream truncated;
+  truncated << s.substr(0, maxlen);
```
I don't know whether to care too much since this is just for logging, but it strikes me as odd that the string returned can actually be maxlen+len(hash) characters long.  I'd rather guarantee getting back something that fits in maxlen.

bertmaher reviewed (approved):
Lgtm, just a couple minor comments, address them if you wish :-)

bertmaher commented on torch/csrc/jit/passes/utils/subgraph_utils.cpp:
```diff
@@ -394,6 +394,31 @@ Node* createSingletonSubgraphAndUpdateAliasing(
+}
+
+std::string generateNameForGraph(
+    std::shared_ptr<Graph> graph,
```
Oh also the lint about shared_ptr<Graph> -> const Graph& is probably a good idea.

Commit:
Update on "[JIT] SubgraphUtils: add a function for generating a string name for a given graph."


The function simply goes over all aten nodes in the graph and
concatenates their names, truncating the final name to a given length.

Differential Revision: [D24698272](https://our.internmc.facebook.com/intern/diff/D24698272)

[ghstack-poisoned]

Commit:
Update on "[JIT] SubgraphUtils: add a function for generating a string name for a given graph."


The function simply goes over all aten nodes in the graph and
concatenates their names, truncating the final name to a given length.

Differential Revision: [D24698272](https://our.internmc.facebook.com/intern/diff/D24698272)

[ghstack-poisoned]
